cpp/ql/lib/semmle/code/cpp/models/interfaces/PartialFlow.qll

```diff
@@ -0,0 +1,28 @@
+/**
+ * Provides an abstract class to override the implicit assumption that a
+ * dataflow/taint-tracking model always fully override the parameters they are
+ * are modeled as writing to. To use this QL library, create a QL class
+ * extending `PartialFlowFunction` with a characteristic predicate that selects
+ * the function or set of functions you are modeling and override the
+ * `isPartialWrite` predicate.
+ *
+ * Note: Since both `DataFlowFunction` and `TaintFunction` extend this class
+ * you don't need to explicitly add this as a base class if your QL class
+ * already extends either `DataFlowFunction` or `TaintFunction`.
+ */
+
+import semmle.code.cpp.Function
+import FunctionInputsAndOutputs
+import semmle.code.cpp.models.Models
+
+/**
+ * A function that may (but not always) updates (part of) a `FunctionOutput`.
+ */
+abstract class PartialFlowFunction extends Function {
+  /**
+   * Holds if the write to `output` either is:
+   * - Only partially updating the `output`
+   * - Is not unconditional
```

Comment:
Yeah, I had a hard time fully defining this since the thing I really want to write is:
> Holds if the write in most cases overrides so much of the buffer that there should not be use-use flow out of the argument.

But that's not a very clear description 😂

Reply:
How about:
> Holds if the write to `output` does not overwrite the entire value that was there before, or does not do so reliably. For example the destination argument of `strcat` is modified but not overwritten.

Reply:
To be honest though, the main thing is I don't like the double-negative in "Is not unconditional", so I guess we could just simplify it to "is conditional".

Reply:
I like your suggestion above. I'll use that one if you don't mind!

Reply:
Fixed in aca3970c33e8a4ff4744505e295821463f12c30d